packages/ajax/test/AjaxClass.interceptors-cache.test.js

```diff
@@ -0,0 +1,713 @@
+    newCacheId();
+    expect(() => {
+      const indexes = addCacheInterceptors(ajax, {
+        useCache: 'always',
+        // @ts-ignore needed for test
+        timeToLive: '',
+      });
+      removeCacheInterceptors(ajax, indexes);
+    }).to.throw();
+  });
+
+  it('validates cache identifier function', () => {
+    // @ts-ignore needed for test
+    cacheId = '';
+
+    const indexes = addCacheInterceptors(ajax, { useCache: 'always' });
+
+    return ajax.get('/test').catch(
+      /** @param {Error} err */ err => {
+        expect(err.message).to.equal('getCacheIdentifier returns falsy');
+
+        removeCacheInterceptors(ajax, indexes);
+      },
+    );
+  });
+
+  //--
+
+  it("throws when using methods other than `['get']`", () => {
+    newCacheId();
+
+    expect(() => {
+      const indexes = addCacheInterceptors(ajax, {
+        useCache: 'always',
+        methods: ['get', 'post'],
+      });
+      removeCacheInterceptors(ajax, indexes);
+    }).to.throw(/not yet supported/);
+  });
+
+  it('creates a wrapped ajax object', () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, { useCache: 'always' });
+
+    ['delete', 'get', 'head', 'options', 'post', 'put', 'patch'].forEach(method => {
+      expect(ajax[method]).to.be.ok;
+    });
+    removeCacheInterceptors(ajax, indexes);
+  });
+
+  it('preserves ajax configuration', () => {
+    newCacheId();
+
+    /** @type {GlobalCacheOptions} */
+    const cacheParams = {
+      useCache: 'always',
+      methods: ['get'],
+      timeToLive: 3600001,
+      requestIdentificationFn: () => '',
+    };
+    const indexes = addCacheInterceptors(ajax, cacheParams);
+    const ajaxOptions = ajax.options;
+    const ajaxProxyGetSpy = spy(ajax.proxy, 'get');
+
+    return ajax
+      .get('/test')
+      .then(() => {
+        const configArg = ajaxProxyGetSpy.args[0][1];
+        expect(ajaxProxyGetSpy).to.be.calledOnceWith('/test');
+        expect(JSON.stringify(configArg)).to.be.equal(
+          JSON.stringify({
+            ...ajaxOptions,
+          }),
+        );
+      })
+      .finally(() => {
+        ajaxProxyGetSpy.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it("returns the cached object on second call with `useCache: 'always'`", () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 100,
+    });
+    const ajaxGetSpy = sinon.spy(ajax, 'get');
+
+    return ajax
+      .get('/test')
+      .then(() => {
+        expect(ajaxGetSpy.calledOnce).to.be.true;
+        expect(ajaxGetSpy.calledWith('/test')).to.be.true;
+      })
+      .then(() => ajax.get('/test'))
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('all calls with non-default `timeToLive` are cached proactively', () => {
```

Comment:
```suggestion
  it('all calls with non-default `timeToLive` are cached on demand per request', () => {
```

Or maybe I don't understand the behaviour...

If timeToLive is kept as a default, but then we do:

```js
ajax.get('/test', {
  lionCacheOptions: {
    useCache: 'always',
  },
}),
```

Do we get a cached response with default timeToLive expiration time? Or do we not get a cached response because timeToLive is still default value?

Reply:
The behaviour is to cache by default and use the cache only when needed, by using useCache=true.
So the app could collect responses in cache but not use it, and only when , for instance, a component does a call with useCache=true (and cache is not expired), the component will get a response from cache.
This behaviour is aligned with @LarsDenBakker some time ago